src/index.ts

```diff
@@ -21,38 +24,54 @@ function makeBase58CheckEncoder(p2pkhVersion: number, p2shVersion: number): (dat
         ) {
           throw Error('Unrecognised address format');
         }
-        addr = Buffer.concat([Buffer.from([p2pkhVersion]), data.slice(3, 3 + data.readUInt8(2))]);
+        if (p2pkhVersion[0] instanceof Array) {
```

Comment:
Why do we have an array if we ignore all elements but the first?

Reply:
I didn't write this part but I am assuming that the original contributor responded to your comment where you asked him to refactor to take `either a buffer or an array of bytes, and check for the version inside that array`, hence it's checking which pattern it is.

https://github.com/ensdomains/address-encoder/pull/10#discussion_r349373290